internal/injector/aspect/advice/code/dot_function.go

```diff
@@ -45,6 +45,8 @@ type (
 		// LastResultThatImplements returns the name of the last return value in this function that implements
 		// the provided interface type, or an empty string if none is found.
 		LastResultThatImplements(string) (string, error)
+		// FinalResultImplements returns whether the final result implements the provided interface type.
+		FinalResultImplements(string) (bool, error)
```

Comment:
It's odd that this is `FinalResult` but just above is `LastResult`?
I'd use `Last` for both...